forui/lib/src/widgets/select_menu_tile.dart

```diff
@@ -222,6 +258,142 @@ class FSelectMenuTile<T> extends FormField<Set<T>> with FTileMixin {
           },
         );
 
+  /// Creates a [FSelectMenuTile] that lazily builds the menu.
+  ///
+  /// The [menuTileBuilder] is called for each tile that should be built. It will be called only for indices <= [count]
+  /// if [count] is specified.
+  ///
+  /// The [predicate] returns true if the tile at the given index should be built. It may be called more than once for
+  /// the same index.
+  ///
+  /// The [count] is the number of tiles to build. If null, [menuTileBuilder] will be called until [predicate] returns
+  /// false.
```

Comment:
Ah, flutter builder doesn't automatically calculate this based on what is visible on the viewport?

Reply:
I think there's some slight misunderstanding. These arguments are to allow fine-grained control over how many tiles are built before the list ends.

Reply:
Count does make sense to me, but I'm not too sure about the predicate. When will we ever want to not build a certain tile.

Reply:
Either a predicate or count can be provided. It mirrors [ListView.builder](https://api.flutter.dev/flutter/widgets/ListView/ListView.builder.htmll) allowing the builder to return null. In our case we can't do that due to implementation details. We allow people to pass in a predicate as an an alternative to the builder returning null.

Reply:
I don't see a predicate function as a parameter in `Listview.builder` though.

https://api.flutter.dev/flutter/widgets/ListView/ListView.builder.html

Reply:
> So if the predicate returns false, the element will be skipped?
Yes, it will skip the element if it returns false.

> I don't see a predicate function as a parameter in `Listview.builder` though.
> 
> https://api.flutter.dev/flutter/widgets/ListView/ListView.builder.html
The predicate function is an alternative to `Listview.Builder` returning null. We cannot return null due to implementation constraints.

Reply:
I see, why don't we accept null for the builder function? Could be better to follow flutter's implementation